protocol/v2/qbft/controller/controller.go

```diff
@@ -151,6 +151,7 @@ func (c *Controller) UponExistingInstanceMsg(logger *zap.Logger, msg *specqbft.P
 		logger.Debug("❌ failed to broadcast decided message", zap.Error(err))
 	}
 
+	// TODO: spec checks if prevDecided (and returns if so) right after inst.ProcessMsg. Should we align?
```

Comment:
can you explain further? we also check here for prevDecided.. whats the diff?

Reply:
@nkryuchkov is it on purpose on just not aligned?

Reply:
I only applied what was changed since the previous alignment: https://github.com/ssvlabs/ssv-spec/compare/tmp-160724-spec-align...alan?expand=1

I didn't touch the rest because I don't know if it's intentional or not.

Regarding this particular change, it looks to me like we need to align it but carefully test it afterwards.

Reply:
the reason the implementation is returning later here is most likely so it can broadcast multiple decideds (and not just the first one at quorum), in order for the exporter to more reliably tell whether all operators signed or not.

since the exporter doesn't longer use decided messages to determine participation, then this misalignment should no longer be needed.

Reply:
> the reason the implementation is returning later here is most likely so it can broadcast multiple decideds (and not just the first one at quorum), in order for the exporter to more reliably tell whether all operators signed or not.
> 
> since the exporter doesn't longer use decided messages to determine participation, then this misalignment should no longer be needed.
> 
> we can align this to spec imo

@nkryuchkov

Reply:
done

Reply:
@moshe-blox fixed it